src/tools/compiletest/src/header.rs

```diff
@@ -982,3 +953,36 @@ fn parse_normalization_string(line: &mut &str) -> Option<String> {
     *line = &line[end + 1..];
     Some(result)
 }
+
+// Takes a directive of the form "<version1> [- <version2>]",
+// returns the numeric representation of <version1> and <version2> as
+// tuple: (<version1> as u32, <version2> as u32)
+// If the <version2> part is omitted, the second component of the tuple
+// is the same as <version1>.
+fn extract_version_range<F>(line: &str, parse: F) -> (u32, u32)
+where
+    F: Fn(&str) -> Option<u32>,
+{
+    const ERROR_MESSAGE: &'static str = "Malformed GDB version directive";
```

Comment:
This doesn't need to be a const -- and the message is misleading as this is used for more than just GDB versions now.

Reply:
Copied it to `.expect` calls.